Amazon.IonDotnet/Internals/Text/IonTextWriter.cs

```diff
@@ -240,25 +275,46 @@ public override void WriteBlob(ReadOnlySpan<byte> value)
 
             var base64 = Convert.ToBase64String(value.ToArray());
 
-            this.textWriter.Write("{{");
-            if (this.options.PrettyPrint)
+            if (this.options.BlobAsString)
             {
-                this.textWriter.Write(' ');
+                this.textWriter.Write('"');
+            }
+            else
+            {
+                this.textWriter.Write("{{");
+
+                if (this.options.PrettyPrint)
+                {
+                    this.textWriter.Write(' ');
+                }
             }
 
             this.textWriter.Write(base64);
-            if (this.options.PrettyPrint)
+
+            if (this.options.BlobAsString)
             {
-                this.textWriter.Write(' ');
+                this.textWriter.Write('"');
             }
+            else
+            {
+                if (this.options.PrettyPrint)
+                {
+                    this.textWriter.Write(' ');
+                }
 
-            this.textWriter.Write("}}");
+                this.textWriter.Write("}}");
+            }
 
             this.CloseValue();
         }
 
         public override void WriteClob(ReadOnlySpan<byte> value)
         {
+            if (this.options.JsonDowngrade)
+            {
+                throw new NotSupportedException("JSON downgrading Clob is not yet supported");
+            }
```

Comment:
Note: this is probably where that unicode escaping method you found would come in handy.